### PR TITLE
feat: upgrade to gatsby v3, icon font, better image performance

### LINKED ISSUE #247: Flash of unstyled content on page load in safari
**Describe the bug**
Icons, for example the one inside our CTA, are displayed as dark on page load before the color gets passed to it on safari.

**To Reproduce**
load the page in safari with a clean cache and see what happens to the icons.
_Note: this is not reproducible in development for me, only in deployment._

Screenshot 2021-06-23 at 10 11 25

**Expected behavior**
Icons are displayed correctly on load

**Environment**
Safari `Version 14.1.1 (16611.2.7.1.4)`

**Additional context**
Flash of unstyled content ([wikipedia](https://en.wikipedia.org/wiki/Flash_of_unstyled_content))

### LINKED ISSUE #268: Move away from SVG icons to icon font
**Is your feature request related to a problem? Please describe.**

Moving to an icon-font will resolve several issues:

- #247
- https://github.com/ory/web/issues/245 (partially)

### LINKED ISSUE #278: Migrate to Gatsby v3
**Is your feature request related to a problem? Please describe.**

Many of page insight issues stem from missing code splitting and other webpack 5 features. Upgrading to gatsby v3 should resolve that.

### PR DESCRIPTION
**Pull request by aeneasr**
This PR upgrade gatsby to v3. As part of these changes, several things - including loading times and image loading - have significantly improved.

Closes #278
Closes #268
Closes #247

### CODE REVIEW

**Comment by aeneasr** [comment]
Left = old, right = new

Bildschirmfoto 2021-07-08 um 22 58 01

**Comment by aeneasr** [comment]
Took a quick look at the icons. Phosphor icons actually do support icon fonts (search for "phosphor icon font" and open https://github.com/phosphor-icons/phosphor-icons). Implemented something basic, loads much faster and has no flashing.

File: src/pages/index.tsx [hunk 1]
```diff
@@ -19,55 +18,21 @@ import SEO from '../components/layouts/seo/seo'
...
-  )
-  const UserPlus = (
-    <IconWrapper
-      color={'themed-primary'}
-      icon={'UserPlusDuotone'}
-      size={'32'}
-    />
+    <i className="ph-arrow-right-bold themed-primary size16" />
   )
+  const Terminal = <i className="ph-terminal themed-primary size32" />
```
Comment (aeneasr) [inline src/pages/index.tsx:29]: phosphor icon font

**Comment by aeneasr** [comment]
**old**

Bildschirmfoto 2021-07-09 um 00 08 57
Bildschirmfoto 2021-07-09 um 00 08 55

---

**new**: this will become even faster if deployed not as a preview due to the way netlify cdn works

Bildschirmfoto 2021-07-09 um 00 08 48
Bildschirmfoto 2021-07-09 um 00 08 47

**Comment by aeneasr** [comment]
@jfcurran duotone weight font icons are currently not supported. I think it looks ok but you have to decide:

Bildschirmfoto 2021-07-09 um 10 06 23

**Comment by aeneasr** [comment]
Ok, I got rid of all the CSS order warnings (which can be ignored as we use css modules). Also, I updated all the icons to be font-based icons. Please note that duotone is not available so I had to use the regular icons instead.

However, whenever duotone was used, I added the duotone placeholder class, in case duotone is going to be supported on day in phosphor icon font, e.g.:

```
  <i className="ph-user-gear ph-user-gear-duotone themed-primary size32" />
```

Please @vinckr @jfcurran review this PR and check every page if the icons are loaded properly and everything is working. Thank you!

As part of this PR I also introduced import ordering which I used to potentially resolve the CSS warnings but that didn't help. I kept it around because I think it helps organize the file. That's where most of the merge conflicts will come for your pr @jfcurran

**Comment by jfcurran** [comment]
waiting for netlify like 
giphy (6)

**Comment by aeneasr** [comment]
safari...


https://user-images.githubusercontent.com/3372410/125050622-693f1e00-e0a2-11eb-9d60-cbbe688f4947.mov

**Comment by jfcurran** [comment]
is the safari clip on load or on scroll?

**Comment by jfcurran** [comment]
Regarding the latest build, Icons are not being displayed correctly as the font does not get loaded. 

Screenshot 2021-07-09 at 10 44 55

the network log gives the following:
```
60e80b5e92261690781aeabe--wonderful-sinoussi-8532a7.netlify.app/:1 Refused to load the font 'https://unpkg.com/phosphor-icons@1.3.2/src/font/Phosphor.ttf?65my2u=' because it violates the following Content Security Policy directive: "default-src 'self' blob: 'unsafe-inline' 'unsafe-eval' *.algolianet.com *.algolianet.net storage.googleapis.com github.com *.githubusercontent.com gh-card.dev *.ory.sh www.youtube.com www.youtube-nocookie.com fonts.googleapis.com fonts.gstatic.com data: s.ytimg.com *.cloudfront.net". Note that 'font-src' was not explicitly set, so 'default-src' is used as a fallback.
´´´

**Comment by aeneasr** [comment]
Ah nice catch, thanks

The safari clip is on load. If you defocus safari, it goes away. I couldn't find any issues for it in gatsby. Pretty sure it has some clipping render issues with the white overlay. I just replaced it with a regular img tag so this should be resolved now.

**Comment by aeneasr** [comment]
Icons still not working, I'll take a look later

**Comment by jfcurran** [comment]
Very nice, no FOUC! 🤝

**Comment by aeneasr** [comment]
so ok to merge?

**Comment by aeneasr** [comment]
did you check all the sites / icons?

**Comment by vinckr** [comment]
Checking icons & pages now on my fleet of browsers.

**Comment by vinckr** [comment]
Looks & works great on
- FF 
- Safari 13 ( no glitches in nav, images!)
- Opera
- Vivaldi
- Brave

The only thing I noticed the youtube thumbnail on the blog not working on all browsers, some just show the play button and a white preview.

**Comment by aeneasr** [comment]
Which browsers in particular?

**Comment by vinckr** [comment]
Just checked again, on all of them, it looks like:
asda

File: src/components/blog-hero.tsx [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
 import Container from './freestanding/containers/container'
 import Grid from './freestanding/containers/grid'
-import { pb16, pb24, pb32, pb48 } from './freestanding/utils/padding.module.css'
 import ContentText from './freestanding/content/content-text'
 import Molecule from './freestanding/molecule/molecule'
 
+import { blogHero } from './blog-hero.module.css'
+import { pb16, pb24, pb32, pb48 } from './freestanding/utils/padding.module.css'
+
+import foo from '../images/aeneas.png'
```
Comment (vinckr) [inline src/components/blog-hero.tsx:13]: I assume this was just for testing?

**Review by vinckr — APPROVED** [review]

File: src/components/blog-hero.tsx (hunk 2 above)
Comment (aeneasr) [reply src/components/blog-hero.tsx:13]: 👍

**Comment by aeneasr** [comment]
Ah yes, another CSP issue! :D

**Comment by aeneasr** [comment]
resolved :)

File: package.json [hunk 3]
```diff
@@ -2,103 +2,109 @@
...
-  "prettier": "ory-prettier-styles",
   "config": {
-    "prettierTarget": "{src/**,src}/*.{md,mdx,json,js,css,html,js,ts,tsx}"
+    "prettierTarget": "{src/**/,src/}*.{md,mdx,json,js,css,html,js,ts,tsx}"
   },
   "dependencies": {
     "@babel/plugin-transform-spread": "^7.12.1",
     "@mdx-js/mdx": "~1.6.21",
     "@mdx-js/react": "~1.6.21",
+    "@mediacurrent/gatsby-plugin-silence-css-order-warning": "^1.0.0",
```
Comment (jfcurran) [inline package.json:12]: nice

File: src/components/layouts/quotes/quotes.tsx [hunk 4]
```diff
@@ -1,12 +1,15 @@
-import React from 'react'
+// requires a loader
```
Comment (jfcurran) [inline src/components/layouts/quotes/quotes.tsx:1]: what exactly?

File: src/components/newsletter.module.css [hunk 5]
```diff
@@ -1,8 +1,3 @@
-@import '../styles/properties.css';
-@import '../styles/grid.css';
-@import '../styles/global.css';
-@import '../styles/typography.css';
-
 .newsletter {
```
Comment (jfcurran) [inline src/components/newsletter.module.css:1]: so clean, nice

File: src/pages/index.tsx [hunk 6]
```diff
@@ -100,13 +65,11 @@ const IndexPage = () => {
           </Button>
         }
         image={
-          <img
-            width={'100%'}
-            height={'100%'}
+          <StaticImage
```
Comment (jfcurran) [inline src/pages/index.tsx:68]: this is great

File: src/pages/markdown/tos.md [hunk 7]
```diff
@@ -113,16 +113,15 @@ any implied warranties.
...
-affiliates, officers, directors, employees, consultants, agents, successors and
-assigns from any and all third party claims, liability, damages, costs or
-demands, including, but not limited to, attorneys' fees, arising from (i) your
-use of the Service, including, but not limited to, all content therein and any
-products or services obtained by you through the Service, (ii) the violation of
-these Terms of Service by you, (iii) the infringement by you (or other user of
-the Service using your account) of any intellectual property or other right of
-any person or entity; or (iv) your violation of any applicable law or
-regulation.
+You agree to indemnify, defend and hold harmless ORY, subsidiaries, affiliates,
```
Comment (jfcurran) [inline src/pages/markdown/tos.md:116]: had to look up indemnify, good to know this is an actual word.
Is capitalisation of ORY required here?

File: src/styles/global.css.d.ts [hunk 8]
```diff
@@ -1,5 +1,6 @@
 declare const styles: {
   readonly responsive: string
+  readonly crisp: string
```
Comment (jfcurran) [inline src/styles/global.css.d.ts:3]: thank you for adding this - i believe I forgot to do this last PR

**Review by jfcurran — APPROVED** [review]
Very good, I appreciate the clean up of all import statements, and their hierarchical ordering - we'll make sure to keep it this way from now on.
I will look into the gatsby docs, regarding the correct use of pageQuery. This caught my eye:
```
new
export const pageQuery = graphql`
  query ($path: String!) {
    mdx(frontmatter: { path: { eq: $path } }) {
      body
      frontmatter {
```
vs
```
old
export const pageQuery = graphql`
  query($path: String!) {
    mdx(frontmatter: { path: { eq: $path } }) {
      body
      frontmatter {
```
where you added a space or removed the query (in `team.tsx`)
```
const Team = () => {
  const data = useStaticQuery(graphql`
    {
      aeneas: file(relativePath: { eq: "aeneas.png" }) {
        childImageSharp {
          gatsbyImageData(width: 90, height: 90, layout: FIXED)
```
thank you hackerman!

File: src/components/layouts/quotes/quotes.tsx (hunk 4 above)
Comment (aeneasr) [reply src/components/layouts/quotes/quotes.tsx:1]: This was there before:

```
import 'react-responsive-carousel/lib/styles/carousel.min.css' // requires a loader
```

but got changed by the import reordering

File: src/components/newsletter.module.css (hunk 5 above)
Comment (aeneasr) [reply src/components/newsletter.module.css:1]: Yeah this should have never been included here in the first place

File: src/pages/markdown/tos.md (hunk 7 above)
Comment (vinckr) [reply src/pages/markdown/tos.md:116]: I am not sure, I think its still written in the "old" style.

Comment (aeneasr) [reply src/pages/markdown/tos.md:116]: I did not change anything here, it's just npm run format. Please discuss this in another issue or pr or with @vinckr

File: src/styles/global.css.d.ts (hunk 8 above)
Comment (aeneasr) [reply src/styles/global.css.d.ts:3]: `npm run tcm` or `npm run tcm:watch` to autogenerate these ;)

File: src/pages/index.tsx (hunk 6 above)
Comment (aeneasr) [reply src/pages/index.tsx:68]: absolutely!

**Comment by aeneasr** [comment]
The spaces have no meaning, probably done by prettyfier!

> Very good, I appreciate the clean up of all import statements, and their hierarchical ordering - we'll make sure to keep it this way from now on.

Just run "npm run format" and it will do it automatically :)

**Pull request merged**